.github/scripts/vercel-deploy.sh

```
@@ -0,0 +1,14 @@
+#!/bin/bash
+set -euo pipefail
+
+# Usage: vercel-deploy.sh <environment> <alias> [--prod]
+ENVIRONMENT="$1"
+ALIAS="$2"
+PROD_FLAG="${3:-}"
+
+vcl() { npx --yes vercel@52.0.0 -A apps/dapp/vercel.json "$@"; }
+vcl pull --yes --environment="$ENVIRONMENT"
+vcl build $PROD_FLAG
```

Shellcheck (0.11.0)

[info] Double quote to prevent globbing and word splitting.

(SC2086)